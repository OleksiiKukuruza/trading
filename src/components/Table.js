import PropTypes from 'prop-types';
import styled from 'styled-components';

const Table = styled.div`
  border: 1px solid ${props => props.theme.primary.main};
`;

const Row = styled.div`
  display: flex;

  :not(:last-child) {
    border-bottom: 1px solid ${props => props.theme.background};
  }
`;

const HeaderRow = styled(Row)`
  background-color: ${props => props.theme.background};
`;

const Cell = styled.div`
  flex: ${props => props.size || 1};
  display: flex;
  justify-content: ${props => props.justifyContent || 'flex-start'};
`;

Cell.propTypes = {
  size: PropTypes.number,
  justifyContent: PropTypes.string
};

export { Table, Row, HeaderRow, Cell };
