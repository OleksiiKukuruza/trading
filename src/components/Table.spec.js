import React from 'react';
import { shallow } from 'enzyme';
import { Table, Row, HeaderRow, Cell } from './Table';
import theme from '../theme';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('Table', () => {
  it('renders Table without crashing', () => {
    const { wrapper } = setup(Table, { theme });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Row without crashing', () => {
    const { wrapper } = setup(Row, { theme });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders HeaderRow without crashing', () => {
    const { wrapper } = setup(HeaderRow, { theme });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Cell without crashing', () => {
    const { wrapper } = setup(Cell, { theme });
    expect(wrapper).toMatchSnapshot();
  });
});
