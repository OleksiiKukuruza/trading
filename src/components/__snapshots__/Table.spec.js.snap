// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders Cell without crashing 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
/>
`;

exports[`Table renders HeaderRow without crashing 1`] = `
.c0 {
  background-color: #f1f4f5;
}

<styled.div
  className="c0"
  theme={
    Object {
      "background": "#f1f4f5",
      "primary": Object {
        "contrastText": "#fff",
        "main": "#54b6c8",
      },
      "spacingUnit": 8,
    }
  }
/>
`;

exports[`Table renders Row without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0:not(:last-child) {
  border-bottom: 1px solid #f1f4f5;
}

<div
  className="c0"
/>
`;

exports[`Table renders Table without crashing 1`] = `
.c0 {
  border: 1px solid #54b6c8;
}

<div
  className="c0"
/>
`;
